test/apps/iodemo/io_demo.cc

```diff
@@ -1975,6 +1977,16 @@ class DemoClient : public P2pDemoCommon {
             LOG << "invalid address " << server_addr;
             abort();
         }
+        if (opts().src_addr != NULL) {
+            memset(&local_addr, 0, sizeof(local_addr));
+            local_addr.sin_family = AF_INET;
+            int ret = inet_pton(AF_INET, opts().src_addr, &local_addr.sin_addr);
+            if (ret != 1) {
+                LOG << "invalid address " << opts().src_addr;
+                abort();
+            }
+            src_addr = &local_addr;
```

Comment:
Why do we need src_addr? We can only use local_addr.

Reply:
We need src_addr which is a pointer for function UcxConnection::connect() to know which case has NULL src_addr.

Reply:
Set src_addr explicitly to NULL in else for better readable.